### ISSUE TITLE
remove() from tray menu doesn't work

### ISSUE THREAD

**Issue by elig0n** [hluk/CopyQ#1430]
I have `remove(row)` run from an action that is set as "Global shortcut" and appears on tray menu which doesn't remove the row.
But if I set the action as "In Menu" and use the toolbar button it does. I take it has to do with the window visibility?
How can I remove a row from "Global shortcut" only which runs from the tray menu ?

Thanks

**Comment by hluk** [comment]
Can you paste the whole commands?

**Comment by elig0n** [comment]
copyq:
    row = currentItem()
    if (row != -1) {
    	remove(currentItem())
    }
    
    select(row)
    next()

**Comment by hluk** [comment]
[currentItem()](https://copyq.readthedocs.io/en/latest/scripting-api.html#int%20currentItem) works only when triggered from main window (from menu, application shortcut or Action dialog) - see [Selected Items](https://copyq.readthedocs.io/en/latest/scripting-api.html#selected-items).

**Comment by hluk** [comment]
I'm not sure if I understand. Are you trying to select an item in the tray menu and trigger the global shortcut to remove it?

**Comment by elig0n** [comment]
@hluk I want to remove the currently selected item and select the next one from the tray menu.

**Comment by beefeater7** [comment]
@elig0n You refer to in-window selection, right?
![image](https://user-images.githubusercontent.com/10468970/87231203-5a515d00-c3b5-11ea-83ad-a2e18be0f6ff.png)
[4] in this case

So you...
1. Open main window
2. Select item
3. Open tray menu
4. Run the command?

Because selecting an item IN the tray, then the command also from tray, is obviously not possible...
![image](https://user-images.githubusercontent.com/10468970/87231209-6c330000-c3b5-11ea-966c-12d7c3732a11.png)
You could make a case, that this line [1] is a sort of `currentItem` or `selectedItem` but, for now it isn't. 
If it was, you could maybe make some hotkeys for the systray menu, but that doesn't sound like your use case? Please clarify.

**Comment by hluk** [comment]
Yeah, that's not how it's supposed to work. The tray menu is not associated with the main window and menus don't have any concept of item selection or current item (only simple select-and-activate).

Instead, you can use an "In Menu" command (not "Global Shortcut" type) and optionally set a shortcut (which can be used from the main window). The command will be accessible from the context menu or Item menu (it can be shown with Menu key or usually Alt+I).

**Comment by elig0n** [comment]
@beefeater7 When selecting an entry in the tray it gets 'marked' by such icon :  ![msedge_1wf3KBpGso](https://user-images.githubusercontent.com/31196036/87240884-2eba8b00-c426-11ea-8822-7a52380a240f.png)
